### DISCUSSION TITLE
UART ports on MMI unit

### DISCUSSION THREAD

**Discussion by abmantis** [Arnold-n/P1P2MQTT#66] [General]
The daikin man-machine interface has a pair of UART interfaces. Is the protocol for those know?
Apparently, X9 and X5 are also used to connect a WLAN Adapter (which is not the same as the WLAN Catridge).

![image](https://user-images.githubusercontent.com/974569/224448149-b9c5dd0f-00e7-4cd1-b556-2f5c465172a5.png)

**Comment by Arnold-n** [comment]
Hi @abmantis,

Would be interesting to try; physically these interfaces look a bit similar to the S21 interface used on indoor AC units.

**Reply by brainrecursion** [reply]
Has anyone investigated these ports yet?
If not I will have a look as could be a great way to control the system.

**Reply by Arnold-n** [reply]
Not that I am aware of, would be interesting!

**Reply by brainrecursion** [reply]
The pinout for both X9/X5 seems to be:
1 - 5v
2 - RX
3 - TX
4 - 12v
5 - GND

Can't see any activity on debug UART but getting `~\0\x10\x83\x07\x06\0\xB9\x93~` repeated over and over on the WLAN UART at 115200 baud.

**Reply by brainrecursion** [reply]
Get much more out with everything connected up. Some examples:
```
~\x01\x1E\x03\x0A\xA1~
~\x01\xB1\x9D\xB2~
~\x01\x10\x03\x1A;~
~\x01\xD1\x9B\xD1~
~\x01\x12\x03\xAA\x08~
~\x01\xF1\x99\xF0~
~\x01\x14\x03z\~
~\x01\x11\x97\x17~
~\x01\x16\x03\xCAo~
~\x011\x956~
~\x01\x18\x03\xDA\xF5~
~\x01Q\x93U~
~\x01\x1A\x03j\xC6~
~\x01q\x91t~
~\x01\x1C\x03\xBA\x92~
~\x01\x91\x9F\x93~
~\x01\x1E\x03\x0A\xA1~
~\x01\xB1\x9D\xB2~
~\x01\x10\x03\x1A;~
~\x01\xD1\x9B\xD1~
~\x8~
~\x01\xF1\x99\xF0~
~\x01\x14\x03z\~
~\x01\x11\x97\x17~
~\x01\xF1\x99\xF0~
~\x10\x10{"hydroUnit.powerInputHeatingHeatPumpAvg4m":{"settable":false,"valid":true,"readonly":true,"requiresReboot":false,"objectKind":"charNumber","value":488.00,"minValue":0.00,"maxValue":31700.00,"stepValue":1.00,"unit":"W"}]}]#5~
~\x10\x12{"hydroUnit.powerOutputHeatingAvg4m":{"settable":false,"valid":true,"readonly":true,"requiresReboot":false,"objectKind":"charNumber","value":1710.00,"minValue":0.00,"maxValue":31700.00,"stepValue":1.00,"unit":"W"}]}]\x89\xA8~
~\x10\x12{"domesticHotWaterTank.dhwTemp":{"settable":false,"valid":true,"readonly":true,"requiresReboot":false,"objectKind":"charNumber","value":45.00,"minValue":-127.00,"maxValue":127.00,"stepValue":1.00,"unit":"C"}]}]\x98\xb3~
```

**Comment by abmantis** [comment]
![20230311_213308](https://user-images.githubusercontent.com/974569/224513598-23a69952-fadf-4ca7-b85b-e2418abcbc6e.jpg)
This is the controller back. I'm not sure what model it is, I've only seen it refered as MMI2, but the heatpump is an Altherma 3 M Monoblock.
